javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -424,93 +424,88 @@ && shouldGeneratePagingMethods()) {
                 JavaSettings.PollingDetails pollingDetails = settings.getPollingConfig(proxyMethod.getOperationId());
 
                 MethodPollingDetails methodPollingDetails = null;
+                MethodPollingDetails dpgMethodPollingDetailsWithModel = null;   // for additional LRO methods
                 if (pollingDetails != null) {
                     methodPollingDetails = new MethodPollingDetails(
                         pollingDetails.getStrategy(),
                         getPollingIntermediateType(pollingDetails, syncReturnType),
                         getPollingFinalType(pollingDetails, syncReturnType),
                         pollingDetails.getPollIntervalInSeconds());
-                    builder = builder.methodPollingDetails(methodPollingDetails);
-                }
 
-                if (settings.getSyncMethods() != JavaSettings.SyncMethodsGeneration.NONE) {
-                    // begin method async
-                    methods.add(builder
-                        .returnValue(createLongRunningBeginAsyncReturnValue(operation, proxyMethod, syncReturnType, methodPollingDetails))
-                        .name("begin" + CodeNamer.toPascalCase(proxyMethod.getSimpleAsyncMethodName()))
-                        .onlyRequiredParameters(false)
-                        .type(ClientMethodType.LongRunningBeginAsync)
-                        .isGroupedParameterRequired(false)
-                        .methodVisibility(methodVisibility(ClientMethodType.LongRunningBeginAsync, false))
-                        .build());
-
-                    if (settings.isContextClientMethodParameter()) {
-                        addClientMethodWithContext(methods,
-                            builder.methodVisibility(methodVisibility(ClientMethodType.LongRunningBeginAsync, true)),
-                            parameters);
+                    if (settings.isLowLevelClient() &&
+                            !(ClassType.BinaryData.equals(methodPollingDetails.getIntermediateType())
+                                    && ClassType.BinaryData.equals(methodPollingDetails.getFinalType()))) {
+                        // a new method to be added as implementation only (not exposed to client) for developer
+                        dpgMethodPollingDetailsWithModel = methodPollingDetails;
+
+                        // DPG keep the method with BinaryData
+                        methodPollingDetails = new MethodPollingDetails(
+                                dpgMethodPollingDetailsWithModel.getPollingStrategy(),
+                                ClassType.BinaryData, ClassType.BinaryData,
+                                dpgMethodPollingDetailsWithModel.getPollIntervalInSeconds());
```

Comment:
Other change in the class is refactor.

1. If DPG, keep the origin client method of `<BinaryData, BinaryData>`, even if polling configuration  provide different intermediate type and final type.
2. If polling configuration provides different intermediate type and final type, generate additional LRO method with suffix "WithModel" in implementation only. This would enable developer to customize the LRO method in client without trying to convert say `PollerFlux<BinaryData, BinaryData>` to `PollerFlux<LroProduct, LroProduct>`